src/runtime_pthread.js

```diff
@@ -90,7 +90,7 @@ if (ENVIRONMENT_IS_PTHREAD) {
   // notified about them.
   self.onunhandledrejection = (e) => { throw e.reason || e; };
 
-  function handleMessage(e) {
+  {{{ asyncIf(ASYNCIFY == 2) }}} function handleMessage(e) {
```

Comment:
It seems like these fixes would apply to JSPI using in the main function of any pthread, not just the PROXY_TO_PTHREAD mode.   Is there something specific to PROXY_TO_PTHREAD here?

Reply:
Yeah, it's not totally specific to PROXY_TO_PTHREAD, more it showed up as an issue there since the entry point for the main proxy is not the users actual main. 

With a user defined entry point there were ways of handling this (see core.test_pthread_join_and_asyncify). I think I can actually change that test now to not require the user to specify that the entry function is JSPI. Hower, this makes me wonder if we should automatically JSPI wrap thread entry functions or require the user to specify that they need JSPI. FWIW, we currently auto JSPI wrap main.

Reply:
Do an alternative solution to this one would be to auto-wrap ` _main_thread` which is used in `_emscripten_proxy_main` ?

If the clang annotation was working we could use that there?

Reply:
> So an alternative solution to this one would be to auto-wrap ` _main_thread` which is used in `_emscripten_proxy_main` ?

Yeah, we'd need `EMSCRIPTEN_KEEPALIVE` or add it to `EXPORTED_FUNCTIONS` and also add it to `DEFAULT_ASYNCIFY_EXPORTS`.

> If the clang annotation was working we could use that there?

Yeah, I still need to fix binaryen to handle the annotation.